$NetBSD: patch-ae,v 1.6 2025/03/30 15:44:42 wiz Exp $

Fix build on OpenBSD.
Calling usleep for 1000000 microseconds or more is not allowed, so add a
myusleep() wrapper to DTRT.

--- src/proxy.h.orig	2021-07-01 16:59:57.000000000 +0000
+++ src/proxy.h
@@ -21,6 +21,9 @@
 #include <ctype.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#if !(defined(__OpenBSD__)
+#include <sys/timeb.h>
+#endif
 #include <fcntl.h>
 #include <time.h>
 
@@ -100,6 +103,8 @@
 #endif
 void daemonize(void);
 #define SLEEPTIME 1000
+#define usleep(usecs) myusleep(usecs)
+int myusleep(useconds_t);
 #ifndef O_BINARY
 #define O_BINARY 0
 #endif
